### PR TITLE
merge pain palette changing for Chex

### PR DESCRIPTION
**Pull request by fabiangreffrath**

### CODE REVIEW

File: prboom2/src/st_stuff.c [hunk 1]
```diff
@@ -791,15 +791,26 @@ static void ST_doPaletteStuff(void)
...
   if (cnt)
     {
-      palette = (cnt+7)>>3;
-      if (palette >= NUMREDPALS)
-        palette = NUMREDPALS-1;
+      // In Chex Quest, the player never sees red.  Instead, the
+      // radiation suit palette is used to tint the screen green,
+      // as though the player is being covered in goo by an
+      // attacking flemoid.
+      if (gamemission == chex)
```
Comment (kraflab) [inline prboom2/src/st_stuff.c:798]: We should still have a check if the pain palette is enabled right?

Comment (fabiangreffrath) [reply prboom2/src/st_stuff.c:798]: If it wasn't, we wouldn't even enter the `if (cnt)` block.

Comment (kraflab) [reply prboom2/src/st_stuff.c:798]: Ah right

**Review by kraflab — APPROVED** [review]
Thanks 🥣

**Pull request merged**